File: Embarcacion.cpp
#include "Embarcacion.h"

Embarcacion::Embarcacion(string n, int c, double l, int a) {
    nombre = n;
    capacidad = c;
    longitud = l;
    anioFabricacion = a;
    cout << "Se ha creado la embarcación: " << nombre << endl;
}

Embarcacion::~Embarcacion() {
    cout << "Se ha destruido la embarcación: " << nombre << endl;
}

void Embarcacion::mostrarDatos() {
    cout << "\n--- DATOS DE LA EMBARCACIÓN ---" << endl;
    cout << "Nombre: " << nombre << endl;
    cout << "Capacidad: " << capacidad << " personas" << endl;
    cout << "Longitud: " << longitud << " metros" << endl;
    cout << "Año de fabricación: " << anioFabricacion << endl;
}

File: Embarcacion.py
class Embarcacion:
    def __init__(self, nombre, tipo, capacidad, longitud):
        self.nombre = nombre
        self.tipo = tipo
        self.capacidad = capacidad
        self.longitud = longitud
        print(f"Embarcación '{self.nombre}' creada.")

    def mostrar_info(self):
        print(f"Nombre: {self.nombre}")
        print(f"Tipo: {self.tipo}")
        print(f"Capacidad: {self.capacidad} personas")
        print(f"Longitud: {self.longitud} metros")

    def navegar(self):
        print(f"La embarcación {self.nombre} está navegando...")

    def __del__(self):
        print(f"Embarcación '{self.nombre}' destruida.")
